server/lib/report_server/clue.ex

```diff
@@ -0,0 +1,207 @@
+        "denormalized" => data.structure
+      }}
+    else
+      error -> error
+    end
+  end
+
+  defp query_for_text_tile_answers(_url, learners,  user = %User{}) do
+    run_remote_endpoints = Enum.map(learners, fn learner -> learner[:run_remote_endpoint] end)
+    sql = get_text_tile_answer_sql(run_remote_endpoints)
+    with {:ok, query_id, _status} <- AthenaDB.query(sql, UUID.uuid4(), user),
+          {:ok, path} <- AthenaQueryPoller.wait_for(query_id) do
+        {:ok, path}
+    else
+      error -> error
+    end
+  end
+
+  defp get_text_tile_answer_sql(run_remote_endpoints) do
+    """
+    WITH last_changes AS (
+      SELECT
+        json_extract_scalar("log1"."parameters", '$.toolId') as tileId,
+        MAX("log1"."time") AS time
+      FROM "log_ingester_qa"."logs_by_time" log1
+      WHERE "log1"."application" = 'CLUE'
+        AND "log1"."event" = 'TEXT_TOOL_CHANGE'
+        AND json_extract_scalar("log1"."parameters", '$.operation') = 'update'
+        AND "log1"."run_remote_endpoint" in #{ReportUtils.string_list_to_single_quoted_in(run_remote_endpoints)}
+      GROUP BY json_extract_scalar("log1"."parameters", '$.toolId')
+    )
+
+    SELECT
+      "log"."username" AS username,
+      json_extract_scalar("log"."parameters", '$.tileTitle') AS tile_title,
+      json_extract_scalar("log"."parameters", '$.documentKey') AS document_key,
+      json_extract_scalar("log"."parameters", '$.documentHistoryId') as document_history_id,
+      json_extract_scalar("log"."parameters", '$.args[0].text') as text_value
+    FROM "log_ingester_qa"."logs_by_time" log
+      JOIN "last_changes" on (
+        "last_changes"."tileId" = json_extract_scalar("log"."parameters", '$.toolId')
+        AND "log"."time" = "last_changes"."time")
+    WHERE "log"."application" = 'CLUE'
+      AND "log"."event" = 'TEXT_TOOL_CHANGE'
+      AND "log"."run_remote_endpoint" in #{ReportUtils.string_list_to_single_quoted_in(run_remote_endpoints)}
+      AND json_extract_scalar("log"."parameters", '$.operation') = 'update'
+      AND json_extract_scalar("log"."parameters", '$.tileTitle') is not null
+      AND json_extract_scalar("log"."parameters", '$.tileTitle') != ''
+      AND json_extract_scalar("log"."parameters", '$.tileTitle') != '<no title>'
+      AND json_extract_scalar("log"."parameters", '$.tileTitle') not like 'Text %'
+    """
+  end
+
+  defp get_parquet_file_path(url, username, resource_link_id) do
+    bucket = System.get_env("ATHENA_REPORT_BUCKET")
+    escaped_url = ReportUtils.escape_url_for_filename(url)
+    # The 'username' field will be something like "user_id@portal_site".
+    [user_id, portal_site] = String.split(username, "@")
+    platform_id = ReportUtils.escape_url_for_filename("https://#{portal_site}")
+    path = "s3://#{bucket}/partitioned-answers/#{escaped_url}/#{platform_id}/#{resource_link_id}/#{user_id}.parquet";
+    FSS.S3.parse(path, config: get_s3_config())
+  end
+
+  defp get_s3_config() do
```

Comment:
You could also use the `runtime.exs` configuration.  In the future we should clean that up (by removing the dead configs from the old athena query handler) and use that here.